lib/dartino/launch_dartino.dart

```diff
@@ -133,34 +142,45 @@ class DartinoLaunch extends Launch {
       pipeStdio(str, subtle: true);
       if (str.startsWith('localhost:')) {
         try {
-          portCompleter.complete(int.parse(str.substring(10)));
+          portCompleter.complete(int.parse(str.substring(10).trim()));
         } catch (e) {
-          pipeStdio('Failed to parse observatory port from "$str"',
+          pipeStdio('Failed to parse observatory port from "$str"\n',
               error: true);
+          portCompleter.complete(null);
         }
       }
     });
     runner.onStderr.listen((str) => pipeStdio('\n$str\n', error: true));
     runner.execStreaming().then((int exitCode) {
-      pipeStdio('debug session exit code is $exitCode', highlight: true);
+      pipeStdio('debug session exit code is $exitCode\n', highlight: true);
       launchTerminated(exitCode, quiet: true);
     });
-    // TODO(danrubel): Wait up to 5 seconds for the observatory port
-    // as a "reasonable" amount of time.
-    // Consider printing a message and letting the user cancel instead.
-    int observatoryPort = await portCompleter.future
-        .timeout(new Duration(seconds: 5), onTimeout: () => null);
+    this.sdk = sdk;
+    int observatoryPort = await portCompleter.future;
     if (observatoryPort == null) {
-      pipeStdio('Failed to determine observatory port', error: true);
+      pipeStdio('Failed to determine observatory port\n', error: true);
       launchTerminated(1, quiet: true);
       return false;
     }
 
     // Connect to the observatory
+    pipeStdio('Connecting observatory to application on device...\n');
     return await ObservatoryDebugger
         .connect(this, 'localhost', observatoryPort)
-        .then((_) {
+        .then((DebugConnection debugger) {
       servicePort.value = observatoryPort;
+
+      // Listen for the first isolate and start the app
+      StreamSubscription<List<DebugIsolate>> subscription;
+      subscription =
+          debugger.isolates.observeMutation((List<DebugIsolate> isolates) {
+        if (isolates.length > 0) {
```

Comment:
isolates.isNotEmpty instead? I tend to prefer the more semantic methods when they make sense; YMMV